src/components/FileRepo.js

```diff
@@ -31,9 +32,25 @@ import { arrangerProjectId } from 'common/injectGlobals';
 import Select from '../uikit/Select';
 import translateSQONValue from 'common/translateSQONValue';
 import { withApi } from 'services/api';
+import { trackUserInteraction, TRACKING_EVENTS } from 'services/analyticsTracking';
 import ArrangerConnectionGuard from './ArrangerConnectionGuard';
 import { config as statsConfig } from './Stats';
 
+const trackFileRepoInteraction = eventData => {
+  if (eventData.label && isObject(eventData.label)) {
+    eventData.label = JSON.stringify(eventData.label);
+  }
```

Comment:
prefer non-mutation, so could do
```
const trackFileRepoInteraction = ({ label, ...eventData }) =>
  trackUserInteraction({
    ...eventData,
    ...(label && { label: isObject(label) ? JSON.stringify(label) : label }),
    category: 'File Repo',
    action: 'default file repo action'
  })
```

Reply:
on this note, I've seen the `JSON.stringify` label thing a couple times. We could do this same `label` handling business in the `trackUserInteraction` function so it doesn't have to be repeated by callers